dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBVectorStoreRecordCollection.cs

```diff
@@ -289,6 +289,35 @@ private async Task CreateIndexAsync(string collectionName, CancellationToken can
         }
     }
 
+    private async Task<IAsyncCursor<BsonDocument>> FindAsync(FilterDefinition<BsonDocument> filter, GetRecordOptions? options, CancellationToken cancellationToken)
+    {
+        ProjectionDefinitionBuilder<BsonDocument> projectionBuilder = Builders<BsonDocument>.Projection;
+        ProjectionDefinition<BsonDocument>? projectionDefinition = null;
+
+        var includeVectors = options?.IncludeVectors ?? false;
+
+        if (!includeVectors && this._vectorProperties.Count > 0)
+        {
+            foreach (var vectorProperty in this._vectorProperties)
+            {
+                var propertyToExclude = this._storagePropertyNames[vectorProperty.DataModelPropertyName];
```

Comment:
Very small optimization: Consider storing the vector storage property names in a field, instead of the vector property objects, that way that the conversion is only done once.

Reply:
Thanks, done!